Tests/PackageGraphTests/DependencyResolverTests.swift

```diff
@@ -551,6 +551,70 @@ class DependencyResolverTests: XCTestCase {
        ])
     }
 
+    func testIncompleteMode() throws {
+        let provider = MockPackagesProvider(containers: [
+            MockPackageContainer(name: "A", dependencies: [
+                "1.0.0": [
+                    (container: "B", requirement: .versionSet(v1Range)),
+                ],
+            ]),
+            MockPackageContainer(name: "B", dependencies: [
+                "1.0.0": [],
+            ]),
+        ])
+
+        let resolver = MockDependencyResolver(provider, MockResolverDelegate())
+        resolver.incompleteMode = true
```

Comment:
I was thinking to have a parameter "debug if unsatisfiable" on the resolve() method. That will initially run the resolver in full mode and then put it in incomplete mode and start the delta debugging algo. The advantage is that the clients who don't want debugging won't get it (like the tests which check for failures). The idea was that the resolver is expected to run in normal mode before getting sent into incomplete mode but I think it should be fine if this is internal set for now. Does that make sense?